python/src/equistore/operations/_utils.py

```
@@ -48,58 +48,119 @@ def _check_blocks(a: TensorBlock, b: TensorBlock, props: List[str], fname: str):
+                raise ValueError(err_msg + err_msg_len)
+
+            for c1, c2 in zip(a.components, b.components):
+                if not (c1.names == c2.names):
+                    raise ValueError(err_msg + err_msg_names)
+
+                if not np.all(c1 == c2):
+                    raise ValueError(err_msg + err_msg_1)
+
         else:
             raise ValueError(
                 f"{prop} is not a valid property to check. "
-                "Choose from 'samples', 'properties', 'components' "
-                "or 'gradients'."
+                "Choose from 'samples', 'properties', 'components'."
             )
 
 
-def _check_same_gradients_components(a: TensorBlock, b: TensorBlock, fname: str):
-    for parameter, grad_a in a.gradients():
-        grad_b = b.gradient(parameter)
-        grad_comps_a = np.array(grad_a.components, dtype=object)
-        grad_comps_b = np.array(grad_b.components, dtype=object)
+def _check_same_gradients(a: TensorBlock, b: TensorBlock, props: List[str], fname: str):
+    """Check if metadata between two gradients's TensorBlocks is consistent
+     for an operation.
 
-        if not np.all(grad_comps_a == grad_comps_b):
-            raise ValueError(
-                f"input to {fname} should habe the same gradient components"
-            )
+    The functions verifies that the metadata of the given props is the same
+    (length, names, values) and in the same order.
+
+    If props is None it only checks if the ``'parameters'`` are consistent.
+
+    :param a: first :py:class:`TensorBlock` for check
+    :param b: second :py:class:`TensorBlock` for check
+    :param props: A list of strings containing the property to check.
+                 Allowed values are ``'samples'`` or ``'properties'``,
+                 ``'components'``. To check only if the ``'parameters'`` are consistent
+                 use ``props=None``, using ``'parameters'`` is allowed
+                  even though deprecated.
```

Why are we keeping around deprecated stuff? We don't have a release yet, we can break everything =)

---

deprecated is maybe not the correct word, the idea is that I wanted to have both behavior: `props=None` -> checks the ``parameters`` and  the possibility to not break the code for someone that writes: `props=['samples',... whatever...,'parameters']`, thinking that `parameters` is needed to check the parameters, which are actually always checked.

---

I would raise an error if someone passes `parameters`, and make it clear in the doc that they are always checked